# Internship Task: Filter Emails

# > Extract Emails from Inbox:

In [1]:
import imaplib
import email
import base64
from email.header import decode_header
from getpass import getpass

username = input('Email: ')
password = getpass(prompt='Password: ', stream=None)

Email: dineshgite77@gmail.com
Password: ········


In [2]:
# create an IMAP4 class with SSL 
imap = imaplib.IMAP4_SSL("imap.gmail.com")
# authenticate
imap.login(username, password)

('OK', [b'dineshgite77@gmail.com authenticated (Success)'])

# > Mention how many emails you want to fetch:
##### That is privide value of "N" : 
##### I am extracting first 1000 emails therefore I take N = 1000

In [3]:
status, messages = imap.select("INBOX")
# number of top emails to fetch
N = 1000
# total number of emails
messages = int(messages[0])

In [4]:
mail_subject = []
mail_from = []
for i in range(messages, messages-N, -1):
    res, msg = imap.fetch(str(i), "(RFC822)")
    for response in msg:
        if isinstance(response, tuple):
            msg = email.message_from_bytes(response[1])
            subject = decode_header(msg['Subject'])[0][0]
            mail_subject.append(subject)
            from_w = msg.get('From')
            mail_from.append(from_w)
            print("Subject:", subject)
            print("From:", from_w)
imap.close()
imap.logout()

Subject: Post-Market Report: Markets snap 10-day gaining streak. Sensex cracks 1066 points
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: b'Dinesh, 15 New Internships Matching Your Profile - Apply Today!'
From: "Internshala" <student@internshala.com>
Subject: sawpna_2605, surve.sahil, and others sent you messages
From: "$w@pn@ on Instagram" <no-reply@mail.instagram.com>
Subject: Critical security alert for your linked Google Account
From: Google <no-reply@accounts.google.com>
Subject: DINESH, you're getting noticed
From: LinkedIn <messages-noreply@linkedin.com>
Subject: Sharekhan POP Technical Report: Momentum Swing - United Breweries Ltd
From: "Sharekhan Technical Research" <no-reply@customercommunication.sharekhan.com>
Subject: Ayush Ruel Das, your new connection,...
From: LinkedIn <notifications-noreply@linkedin.com>
Subject: Stock Update - Infosys, The Ramco Cements Limited
From: "Sharekhan Investor Eye" <no-reply@customercommunication.sharekhan.c

Subject: Talent Corner Hr Services Private Limited is looking for: Data
 Scientist.
From: LinkedIn <jobs-listings@linkedin.com>
Subject: Diversity & Inclusion in Technology [Live Webinar]
From: "Great Learning Academy" <academy@greatlearning.in>
Subject: Post-Market Report: Markets end higher. Smallcap, Midcaps underperform
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: New job - Business Analyst Intern | More jobs matching your Custom Job Alert - Business Analytics Professional Nbfc
From: Naukri Alerts <naukrimyalerts@naukri.com>
Subject: Congratulate Rohan Malkar and 4 others...
From: LinkedIn <notifications-noreply@linkedin.com>
Subject: Job | US healthcare voice process - send cv on email if interested
From: Infinx Services Pvt Ltd <reena.daW5maW54LmNvbQ==@naukri.com>
Subject: DINESH, start a conversation with your new connection, Pratik
From: Pratik Kalsekar via LinkedIn <invitations@linkedin.com>
Subject: Viewpoint - Dixon Technologies Limited; 

Subject: DINESH, you have 2 invitations and 1 new update waiting for you on
 LinkedIn
From: LinkedIn <notifications-noreply@linkedin.com>
Subject: See Sandeep's and other people's connections, experience, and more
From: LinkedIn <invitations@linkedin.com>
Subject: Dinesh Gitaye, Urgent Requirement for the role of Data Executive
From: Naukri Alerts <naukrialerts@naukri.com>
Subject: Congratulate Dhruvi Sutaria for starting...
From: LinkedIn <notifications-noreply@linkedin.com>
Subject: Stock Update - Zydus Wellness Limited;  Viewpoint - Larsen & Toubro Infotech
From: "Sharekhan Investor Eye" <no-reply@customercommunication.sharekhan.com>
Subject: IPO Flash: Mazagon Dock Shipbuilders
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: Post-Market Report: Markets fail to hold opening gains. Nifty holds 11,200
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: surve.sahil, anurag_goku, and others sent you messages
From: =?UTF-8?B

Subject: Post-Market Report: Markets end lower for fourth straight day
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: surve.sahil, sawpna_2605, and others sent you messages
From: =?UTF-8?B?xZrDgkjDjkwgxaDDmlJWw4kgb24gSW5zdGFncmFt?= <no-reply@mail.instagram.com>
Subject: Sharekhan POP Technical Report: Smart Chart - Havells
From: "Sharekhan Technical Research" <no-reply@customercommunication.sharekhan.com>
Subject: Save to the max on your income tax with ELSS
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: See Viraj's and other people's connections, experience, and more
From: LinkedIn <invitations@linkedin.com>
Subject: Dinesh Gitaye, check out the jobs applied by your peers
From: Naukri Alerts <naukrinewalerts@naukri.com>
Subject: Dell XPS 9370 13.3-inch Laptop...
From: "Amazon.in" <store-news@amazon.in>
Subject: Stock Update - HCL Technologies
From: "Sharekhan Investor Eye" <no-reply@customercommunication.sharekhan.c

Subject: aishwarya.sumbre, surve.sahil, and others sent you messages
From: "aishwarya.sumbre on Instagram" <no-reply@mail.instagram.com>
Subject: DINESH, start a conversation with your new connection, Lakshay
From: Lakshay Malhotra via LinkedIn <invitations@linkedin.com>
Subject: Live Sessions from the best! Exclusively for you!
From: GL Academy <gla@academy.greatlearning.in>
Subject: Stock Update - Shree Cement; Viewpoint - Mastek Limited, Amber Enterprises Limited, Solara Active Pharma Sciences
From: "Sharekhan Investor Eye" <no-reply@customercommunication.sharekhan.com>
Subject: Dinesh Gitaye, Urgently hiring for Data Analyst, Trainee
From: Naukri Alerts <naukrialerts@naukri.com>
Subject: dineshgite77, see your 2 new follow requests and more on Instagram
From: "Instagram" <no-reply@mail.instagram.com>
Subject: Job | Hiring Subject Matter Expert | Work From Home
From: Chegg India Pvt Ltd <adilY2hlZ2cuY29t@naukri.com>
Subject: Thermopads Pvt Ltd - ThermoGroup is looking for: Data Scie

Subject: Stock Update - Asian Paints Limited
From: "Sharekhan Investor Eye" <no-reply@customercommunication.sharekhan.com>
Subject: Post Market Report: Global selloff drag Markets lower
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: Explore Intraday Trading with Sharekhan Classroom
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: Check Out Top Companies Hiring Now!
From: Naukri <info@naukri.com>
Subject: Stock Update - Sadbhav Engineering; Viewpoint - Coal India, Astral Poly Technik; Sector Update - Q1FY2021 Cement/Infrastructure/Logistics/Building Material Results Review
From: "Sharekhan Investor Eye" <no-reply@customercommunication.sharekhan.com>
Subject: The Elements of Statistical...
From: "Amazon.in" <store-news@amazon.in>
Subject: Sharekhan Special - Q1FY2021 Results Review
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: Post Market Report: Markets end marginally lower. Nifty hol

Subject: SIP to make every investment count
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: Stock Update - Larsen & Toubro, Bharat Electronics Limited; Viewpoint - City Union Bank
From: "Sharekhan Investor Eye" <no-reply@customercommunication.sharekhan.com>
Subject: Job | Openings for Phone Banking Officer.
From: Job Circle <anuradhaam9iY2lyY2xlLmNvLmlu@naukri.com>
Subject: b'REGULATORY UPDATE \x96 Equity Cash Margin Reporting and Penalties'
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: 5 Paisa Capital is hiring for Marketing Analyst in Mumbai (Andheri) - Apply Soon!!
From: 5 Paisa Capital is Hiring <sonu.sharma@Naukri.com>
Subject: Post Market Report: September series begins on a strong note. Sensex gains 354 points
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: Sharekhan POP Technical Report: Momentum Swing - BIOCON
From: "Sharekhan Technical Research" <no-reply@customercommunicat

Subject: Stock Update - Persistent Systems Limited; Viewpoint - MOIL Limited
From: "Sharekhan Investor Eye" <no-reply@customercommunication.sharekhan.com>
Subject: Post Market Report: Markets edge higher. Sensex up 200 points
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: Your course is waiting for you
From: Great Learning Academy <academy@greatlearning.in>
Subject: Sharekhan POP Technical Report: Momentum Swing - HDFC BANK
From: "Sharekhan Technical Research" <no-reply@customercommunication.sharekhan.com>
Subject: Check Out Top Companies Hiring Now!
From: Naukri <info@naukri.com>
Subject: Stock Update -  Finolex Cables Ltd.
From: "Sharekhan Investor Eye" <no-reply@customercommunication.sharekhan.com>
Subject: Post Market Report: Indices fall on profit booking, weak global cues
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: See Pranay's and other people's connections, experience, and more
From: LinkedIn <invitations@

Subject: Post Market Report: Markets end in green. Pharma stocks in focus
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: b'Classroom Invite \x96 Fundamentals of 3R Research Philosophy'
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: Live Sessions from the best! Exclusively for you!
From: Great Learning <olympus@greatlearning.in>
Subject: b'\xf0\x9f\xa4\x91 BIGGEST DISCOUNTS at Min.90% Off!'
From: =?UTF-8?B?RmxpcGthcnQ=?= <no-reply@ncp.flipkart.com>
Subject: Stock Idea - Pidilite Industries Limited; Stock Update - Mahindra & Mahindra, JK Lakshmi Cement Limited, Cipla, Emami Limited, Lupin, Divis Laboratories; Viewpoint - Honeywell Automation India Limited, Abbott India, Kajaria Ceramics Limited, KEI Industries, Hindustan Petroleum Corporation, Amara Raja Batteries Limited
From: "Sharekhan Investor Eye" <no-reply@customercommunication.sharekhan.com>
Subject: Sharekhan 3R - Stock Idea: Pidilite Industries Limited
From: "

Subject: DINESH, please add me to your LinkedIn network
From: Swaraj Joshi <invitations@linkedin.com>
Subject: Post Market Report: July expiry drags Indian markets lower
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: Help us improve by taking a quick survey!
From: "Naukri.com" <info@naukri.com>
Subject: Sharekhan POP Technical Report: Smart Chart - EQUITAS
From: "Sharekhan Technical Research" <no-reply@customercommunication.sharekhan.com>
Subject: b'SOON IN EFFECT \x96 Equity Cash Margin Reporting Updates and Penalties'
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: Help us improve by taking a quick survey!
From: "Naukri.com" <info@naukri.com>
Subject: Job | Subject Matter Expert |  Statistics |  Work from home
From: Vaidik Eduservices Pvt Ltd <hrrecruiter244@gmail.com>
Subject: Dinesh Gitaye, check out the jobs applied by your peers
From: Naukri Alerts <naukrinewalerts@naukri.com>
Subject: Stock Update - Maruti Suzu

Subject: Job | Urgent opening for QC Chemist with a leading Chemical Manufacturing unit in Taloja.
From: Catalyst Corporate Services Pvt Ltd <careersY2F0c2Vydi5pbg==@naukri.com>
Subject: Check Out Top Companies Hiring Now!
From: Naukri <info@naukri.com>
Subject: JBL Tuner Portable Bluetooth...
From: "Amazon.in" <store-news@amazon.in>
Subject: Stock Update - L&T Technology Services; Viewpoint - Larsen & Toubro Infotech
From: "Sharekhan Investor Eye" <no-reply@customercommunication.sharekhan.com>
Subject: Post Market Report: Markets end 1% higher. Infosys jumps by 10%
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: NSDL Newsletter for Investor Education - The Financial Kaleidoscope - May 2020
From: "NSDL Newsletter" <communications@nsdl.com>
Subject: Sharekhan POP Technical Report: Smart Chart - MOTHERSUMI
From: "SHAREKHAN TECHNICAL RESEARCH" <no-reply@customercommunication.sharekhan.com>
Subject: Dinesh Gitaye, check out the jobs applied by your peers
F

Subject: Shubham Dhamale has 38 new connections
From: LinkedIn <notifications-noreply@linkedin.com>
Subject: Job | Urgent Hiring | SME - Maths | Work from home
From: Vaidik Eduservices Pvt Ltd <hrrecruiter244@gmail.com>
Subject: New job - Manager Business Analyst | More jobs matching your Custom Job Alert - Business Analytics Professional Nbfc
From: Naukri Alerts <naukrimyalerts@naukri.com>
Subject: 2217932 - Digital Retention statement. <{CN20200704150306}>
From: "Sharekhan Retention statement" <digitalcontracts@sharekhan.com>
Subject: b'Hand-picked Stocks to Invest In \x96 Doesn\x92t Get Easier than This!'
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: Stock Idea - Mahanagar Gas Limited
From: "Sharekhan Investor Eye" <no-reply@customercommunication.sharekhan.com>
Subject: Sharekhan 3R - Stock Idea: Mahanagar Gas Limited
From: "Sharekhan 3R Stock Idea" <no-reply@customercommunication.sharekhan.com>
Subject: Post Market Report: Markets on uphill. Nift

Subject: Post Market Report: Markets gain for 3rd day. Glenmark jumps by 27%
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: Congratulate Shubham Dhamale for...
From: LinkedIn <notifications-noreply@linkedin.com>
Subject: Dinesh Gitaye, Handpicked New Jobs for you - Business Operations Analyst
From: Naukri Alerts <naukrialerts@naukri.com>
Subject: Stock Update - Cadila Healthcare Limited, The Ramco Cements Limited, Thermax Limited
From: "Sharekhan Investor Eye" <no-reply@customercommunication.sharekhan.com>
Subject: b'Hand-picked Stocks to Invest In \x96 Doesn\x92t Get Easier than This!'
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: Post Market Report: Markets end higher up by 1.5%. RIL major contributor
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: Check Out Top Companies Hiring Now!
From: Naukri <info@naukri.com>
Subject: Realme X2 Pro (Neptune Blue,...
From: "Amazon.in" <store-n

Subject: Post Market Report: Markets end higher as financial gains. SBI in focus
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: Check Out Top Companies Hiring Now!
From: Naukri <info@naukri.com>
Subject: Shivani Umesh Naik wants to follow you.
From: "Instagram" <no-reply@mail.instagram.com>
Subject: Dinesh Gitaye, New Job Opportunities for Junior Analyst, Business Analyst Trainee, Support Analyst
From: Naukri Alerts <naukrialerts@naukri.com>
Subject: Stock Update - Aurobindo Pharma; Viewpoint - Bharat Petroleum Corporation; Sector Update - Cement
From: "Sharekhan Investor Eye" <no-reply@customercommunication.sharekhan.com>
Subject: Post Market Report: Markets snap 6-day gaining streak. Sensex below 34k
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: Shivani Umesh Naik wants to follow you.
From: "Instagram" <no-reply@mail.instagram.com>
Subject: ETFs: A new Way of Investing - This week on Sharekhan Foresight LIVE
From:

Subject: Dinesh Gitaye, check out the jobs applied by your peers
From: Naukri Alerts <naukrinewalerts@naukri.com>
Subject: Stock Update - HDFC Limited, Bajaj Finserv, Supreme Industries, UPL; Viewpoint - Colgate Palmolive (India), Bosch Limited, JSW Steel
From: "Sharekhan Investor Eye" <no-reply@customercommunication.sharekhan.com>
Subject: Subhash is looking for: Data Manager.
From: LinkedIn <jobs-listings@linkedin.com>
Subject: surve.sahil and aditya.kh sent you messages
From: =?UTF-8?B?xZrDgkjDjkwgxaDDmlJWw4kgb24gSW5zdGFncmFt?= <no-reply@mail.instagram.com>
Subject: Congratulate Shubham Dhamale for...
From: LinkedIn <notifications-noreply@linkedin.com>
Subject: We are back with Week 4 of Foresight LIVE and Classroom sessions!
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: Stock Update - Bajaj Finserv, Supreme Industries; Viewpoint - Colgate Palmolive (India), Bosch Limited, JSW Steel
From: "Sharekhan Investor Eye" <no-reply@customercommunication.sh

Subject: New job - Business Analyst | More jobs matching your Custom Job Alert - Business Analytics Professional Nbfc
From: Naukri Alerts <naukrimyalerts@naukri.com>
Subject: Take care of your investments with our #SafeTodayStrongerTomorrow package
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: Job | Collection Executive
From: DD Service <solution.dd2015@gmail.com>
Subject: Stock Update - Hindustan Unilever Limited
From: "Sharekhan Investor Eye" <no-reply@customercommunication.sharekhan.com>
Subject: Post Market Report: Markets end in green. RIL, HUL, Dr. Reddy in focus
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: Dinesh Gitaye, New Job Opportunities for Data Analyst
From: Naukri Alerts <naukrialerts@naukri.com>
Subject: Job seekers like you are taking these courses
From: Naukri Learning <info@naukri.com>
Subject: Stock Update - HCL Technologies; Viewpoint - RBL Bank
From: "Sharekhan Investor Eye" <no-reply@custome

Subject: Post-Market Report: Weak global cues drags Indian market
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: Stock Update - Infosys
From: "Sharekhan Investor Eye" <no-reply@customercommunication.sharekhan.com>
Subject: Dinesh Gitaye, check out the jobs applied by your peers
From: Naukri Alerts <naukrinewalerts@naukri.com>
Subject: Sony Research India is looking for: Data Scientist.
From: LinkedIn <jobs-listings@linkedin.com>
Subject: Job | WORK FROM HOME IN A TOP BPO. WITH SUPERB SAL.WHATSAPP US NOW
From: "Gratitude India Manpower Consultants Pvt. Ltd." <bangbpoZ3JhdGl0dWRlaW5kaWEuY29t@naukri.com>
Subject: Post-Market Report: Markets end flat. Nifty below 9,300
From: "SHAREKHAN LIMITED" <no-reply@customercommunication.sharekhan.com>
Subject: New job - Assistant Manager - Data Analytics | More jobs matching your Custom Job Alert - Business Analytics Professional Nbfc
From: Naukri Alerts <naukrimyalerts@naukri.com>
Subject: Stock Update - HDFC Bank;

('BYE', [b'LOGOUT Requested'])

# >Data preparation:

In [5]:
import numpy as np
import pandas as pd

In [6]:
dataset = pd.DataFrame({'From': mail_from, 'Subject': mail_subject})

In [7]:
dataset.head()

From  \
0  "SHAREKHAN LIMITED" <no-reply@customercommunic...   
1            "Internshala" <student@internshala.com>   
2  "$w@pn@ on Instagram" <no-reply@mail.instagram...   
3              Google <no-reply@accounts.google.com>   
4           LinkedIn <messages-noreply@linkedin.com>   

                                             Subject  
0  Post-Market Report: Markets snap 10-day gainin...  
1  b'Dinesh, 15 New Internships Matching Your Pro...  
2  sawpna_2605, surve.sahil, and others sent you ...  
3  Critical security alert for your linked Google...  
4                     DINESH, you're getting noticed

##### > Now remove extra text From "From" column: (remove text between '< >' these signs)

In [8]:
def removal(my_list, position):
    return my_list[position]

dataset["From"] = dataset['From'].str.split('<').apply(removal, position=0)

##### > make all the strings in lowercase:
- We are doing this because if we make all the strings in lowercase it is easy to filter out emails using a particular keyword like **'Thank you for applying'**.
- That is emails which have a subject like **"THANK YOU FOR APPLYING"**, "**T**hank **Y**ou **F**or **A**pplying" are treated equally as they converted into **'thank you for applying'**.

In [9]:
dataset["From"] = dataset["From"].str.lower()
dataset["Subject"] = dataset["Subject"].str.lower()

In [10]:
dataset['From'].head()

0      "sharekhan limited" 
1            "internshala" 
2    "$w@pn@ on instagram" 
3                   google 
4                 linkedin 
Name: From, dtype: object

# Filtering out the Emails that are related to JOB:

In [11]:
dataset[dataset.apply(lambda row: row.astype(str).str.contains('job').any(), axis= 1)]

From  \
8                                       naukri alerts    
12                                      naukri alerts    
16                                  mehta enterprises    
18                                 akash vikram kadam    
22                           =?utf-8?b?tglua2vksw4=?=    
..                                                 ...   
988                    future vision career solutions    
992                                     naukri alerts    
996  "gratitude india manpower consultants pvt. ltd."    
997                                     naukri alerts    
998                     freight systems india pvt ltd    

                                               Subject  
8    dinesh gitaye, check out the jobs applied by y...  
12   new job - data analyst / senior data analyst |...  
16   job | (m) your resume has been shortlisted for...  
18   job | (k)your cv has been shortlisted for high...  
22                       get job alerts in your inbox.  
..                                                 ...  
988            job | urgent opening for andheri office  
992  dinesh gitaye, check out the jobs applied by y...  
996  job | aspiring to join a top bpo...whatsapp gr...  
997  new job - data analyst | more jobs matching yo...  
998                        job | hiring for supervisor  

[177 rows x 2 columns]

### > Now, see how many Emails are related to JOB:

In [12]:
Emails_related_to_job = dataset[dataset.apply(lambda row: row.astype(str).str.contains('job').any(), axis= 1)]

In [13]:
Emails_related_to_job['Subject'].count()

177

##### There are 177 emails that are related to job

# > To see all these 177 emails which is related to job:

In [14]:
pd.set_option('display.max_rows', Emails_related_to_job.shape[0]+1)
Emails_related_to_job

From  \
8                                       naukri alerts    
12                                      naukri alerts    
16                                  mehta enterprises    
18                                 akash vikram kadam    
22                           =?utf-8?b?tglua2vksw4=?=    
24                                      naukri alerts    
35                                      naukri alerts    
36                                      naukri alerts    
49                                      naukri alerts    
51                 naukri premium - employer services    
60                                      naukri alerts    
65                                      naukri alerts    
67                            infinx services pvt ltd    
76                                      naukri alerts    
89                                      naukri alerts    
113                                     naukri alerts    
119                           career zone consultancy    
121                                     naukri alerts    
123          "trivium education services (p) limited"    
139                                     naukri alerts    
153                                     naukri alerts    
177                                     naukri alerts    
184                                     naukri alerts    
195                                     naukri alerts    
201    inventurus knowledge solutions private limited    
209                                     naukri alerts    
211                               chegg india pvt ltd    
217                                     career divine    
221                                     naukri alerts    
232  marsh mclennan global services india private l...   
233                                     naukri alerts    
235                                     naukri alerts    
239                                           caliber    
241                                     naukri alerts    
246                                     naukri alerts    
260                               chegg india pvt ltd    
281                            i expert international    
282                            i expert international    
284                                        compliance    
285                                     naukri alerts    
290                                     naukri alerts    
296                                     naukri alerts    
306                                     naukri alerts    
307                                     naukri alerts    
314                                          linkedin    
327                                        compliance    
329                                     naukri alerts    
334                                     naukri alerts    
341                          =?utf-8?b?tglua2vksw4=?=    
344                                 "naukri.com info"    
349                                     naukri alerts    
357                                 "naukri.com info"    
367                            i expert international    
368                                     naukri alerts    
370                              surbhi adarsh ranjan    
372                                            naukri    
376                                        job circle    
390                                     naukri alerts    
396                                     naukri alerts    
400              shunya tattva management consultants    
403                                          linkedin    
406              shunya tattva management consultants    
407                            reliance jio is hiring    
413                                     naukri alerts    
417                                   hiring comrades    
420                            reliance jio is hiring    
424                                     naukri alerts    
428                                     naukri alerts    
443                                     naukri a

## Note:
- These mails are Filter out based on keyword **'job'** because I do not have any mail who have a subject like 'Thank you for applying'. But we can also do it for keyword 'Thank you for applying'.  

- for this, we have to change only one thing in line number **11**. In place of **"job"** we only have to put **"thank you for applying"**.

- That is we have to write code like:

  * dataset[dataset.apply(lambda row: row.astype(str).str.contains(**'thank you for applying'**).any(), axis= 1)]